File: Released/Cora/Objective-C/Managers/TaskManager.m
#import "TaskManager.h"


@implementation TaskManager


+ (TaskManager *)sharedInstance {

	static TaskManager *sharedInstance = nil;
	static dispatch_once_t onceToken;

	dispatch_once(&onceToken, ^{ sharedInstance = [self new]; });

	return sharedInstance;

}


- (void)launchTaskWithArguments:(NSArray *)arguments {

	NSTask *task = [NSTask new];
	NSPipe *outputPipe = [NSPipe pipe];

	[task setLaunchPath:@"/bin/sh"];
	[task setArguments:arguments];
	task.standardOutput = outputPipe;
	[task launchAndReturnError:nil];

	NSData *outputData = [[outputPipe fileHandleForReading] readDataToEndOfFile];
	self.outputString = [[NSString alloc] initWithData:outputData encoding:NSUTF8StringEncoding];

}


@end

File: Cora/Swift/UIKit/Managers/TaskManager.swift
import Foundation


final class TaskManager {

	static let sharedInstance = TaskManager()

	var outputString: String?

	private init() {}

	func launchTask(withArguments arguments: [String]) {

		let pipe = Pipe()
		let task = NSTask()

		task.arguments = arguments
		task.launchPath = "/bin/sh"
		task.standardOutput = pipe
		task.launch()

		let data = pipe.fileHandleForReading.readDataToEndOfFile()
		outputString = String(data: data, encoding: .utf8)

	}

}
